buku

```diff
@@ -3351,18 +3364,23 @@ def import_html(html_soup, add_parent_folder_as_tag, newtag):
             desc = comment_tag.find(text=True, recursive=False)
 
         # add parent folder as tag
-        if add_parent_folder_as_tag:
-            # could be its folder or not
-            possible_folder = tag.find_previous('h3')
-            # get list of tags within that folder
-            tag_list = tag.parent.parent.find_parent('dl')
-
-            if ((possible_folder) and possible_folder.parent in list(tag_list.parents)):
-                # then it's the folder of this bookmark
-                if tag.has_attr('tags'):
-                    tag['tags'] += (DELIM + possible_folder.text)
-                else:
-                    tag['tags'] = possible_folder.text
+        if add_parent_folder_as_tag is not False:
```

Comment:
while this part will be decided by jarun, don't use `is not False` and `is False`

just use simple check (`if not add_parent_folder_as_tag` and `if  use_nested_folder_structure`)

unfortunately i can't find article about when to use/not use `is` keyword but 

- from what i read mostly `is` keyword is used when checking against `None`. see [^1]
- other `if` on buku  don't use `is` to check for boolean value

of course i could be wrong if i miss something here


[^1]: https://stackoverflow.com/a/6575506/1766261

Reply:
@rachmadaniHaryono I agree with using ```is not``` and ```is``` .  These are intended for checking identity of objects. Variable in Python are just references and ```A is B``` is True if both points to the same place in memory. It is simpler to write ```not A``` than, ```A is not False```  in my opinion.  Inappropriate using of checking identity may cause hidden issues:  https://lerner.co.il/2015/06/16/why-you-should-almost-never-use-is-in-python/  https://realpython.com/python-is-identity-vs-equality/  Sorry for log post.